### PR TITLE
Rule Engine Tests

### PR DESCRIPTION
**Pull request by ltabis**
- global rule engine tests
- better rule engine error messages
- removed `Block` status and changed `Continue` to `Next`
- added a helper function to init logs in tests
- `LocalPart` object is now `Indentifier.`
- removed the `run_rule` mendatory function.
```rust
// old syntax
run_rules!(#{
   connect: [
      ...
   ]
   ...
})

// new syntax
#{
   connect: [
      ...
   ]
   ...
}
```
- remapped tokens:
  - `obj` to `object`
  - `object str` to `object string`
  - `object addr` to `object address`
- minor fixes

### CODE REVIEW

File: src/processes/delivery.rs [hunk 1]
```diff
@@ -107,9 +107,9 @@ pub(crate) async fn handle_one_in_delivery_queue(
 
     match result {
         Status::Deny => Queue::Dead.write_to_queue(config, &state.get_context().read().unwrap())?,
-        Status::Block => {
-            Queue::Quarantine.write_to_queue(config, &state.get_context().read().unwrap())?
-        }
+        // Status::Block => {
+        //     Queue::Quarantine.write_to_queue(config, &state.get_context().read().unwrap())?
+        // }
```
Comment (Mathieu-Lala) [inline src/processes/delivery.rs:112]: Remove code

File: src/processes/mime.rs [hunk 2]
```diff
@@ -70,9 +70,9 @@ pub(crate) async fn handle_one_in_working_queue(
 
     match result {
         Status::Deny => Queue::Dead.write_to_queue(config, &state.get_context().read().unwrap())?,
-        Status::Block => {
-            Queue::Quarantine.write_to_queue(config, &state.get_context().read().unwrap())?
-        }
+        // Status::Block => {
+        //     Queue::Quarantine.write_to_queue(config, &state.get_context().read().unwrap())?
+        // }
```
Comment (Mathieu-Lala) [inline src/processes/mime.rs:75]: Remove code

File: src/rules/rule_engine.rs [hunk 3]
```diff
@@ -108,9 +117,6 @@ impl<'a> RuleState<'a> {
             .push("date", "")
             .push("time", "")
             .push("connection_timestamp", std::time::SystemTime::now())
-            .push("metadata", None::<MessageMetadata>)
-            // rule engine's internals.
-            .push("stage", "")
             // configuration variables.
             .push("addr", config.server.addr)
             .push("logs_file", config.log.file.clone())
```
Comment (Mathieu-Lala) [inline src/rules/rule_engine.rs:122]: addr and logs_file are pushed twice ?

**Review by Mathieu-Lala — APPROVED** [review]

File: src/processes/delivery.rs (hunk 1 above)
Comment (ltabis) [reply src/processes/delivery.rs:112]: done in `feat/vsl-body-manipulations#151`

**Pull request merged**